agent/api/task/service_connect.go

```diff
@@ -72,3 +86,33 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ParseServiceConnectAttachment parses the service connect container name and service connect config value
+// from the given attachment.
+func ParseServiceConnectAttachment(scAttachment *ecsacs.Attachment, networkMode string, ipv6Enabled bool) (*ServiceConnectConfig, error) {
+	scConfigValue := &ServiceConnectConfig{}
+	containerName := ""
+
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectConfigKey:
```

Comment:
We are not checking anywhere that these keys are present. Could we validate those?